core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java

```diff
@@ -684,7 +684,8 @@ RangerRazEnabledV4Response rangerRazEnabledInternal(@PathParam("workspaceId") Lo
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = RE_REGISTER_CLUSTER_PROXY_CONFIG, nickname = "reRegisterClusterProxyConfig")
     FlowIdentifier reRegisterClusterProxyConfig(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String crn,
-            @QueryParam("originalCrn") String originalCrn, @QueryParam("initiatorUserCrn") String initiatorUserCrn);
+            @QueryParam("skipFullReRegistration") boolean skipFullReRegistration, @QueryParam("originalCrn") String originalCrn,
```

Comment:
This API and the other below are not exposed publicly, and are only used in a few places my team controls in the Datalake service. So the situation you are referring to would only occur if the Datalake and CB service versions are mismatched.

But I understand where you are coming from, what would be the best way to avoid any potential compatibility issues? We need to move forward with this change to ensure customers do not have issues.

Reply:
For example set a default value for it.

Reply:
I added a default value for it in the API, this will be carried down to the other methods. FYI, a boolean variable in a Spring API method will by default be false, but I have set a default value just to explictly show what is expected.